rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalDataTransferServer.java

```diff
@@ -206,6 +209,7 @@ public void stop()
     SocketChannel localDataTransferClient;
     Configuration conf;
     BookKeeperFactory bookKeeperFactory;
+    int generationNumber;
```

Comment:
Avoid adding this as class member. There is work going on to make this thread long lived and handle multiple requests. In that case, it would be cleaner to have this as a local variable